Code/Framework/AzFramework/AzFramework/Network/IRemoteTools.h

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/base.h>
+#include <AzCore/EBus/Event.h>
+#include <AzCore/Memory/OSAllocator.h>
+#include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/RTTI/ReflectContext.h>
+#include <AzCore/RTTI/RTTI.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/std/containers/deque.h>
+#include <AzCore/std/containers/unordered_map.h>
+#include <AzCore/std/smart_ptr/intrusive_ptr.h>
+#include <AzCore/std/smart_ptr/intrusive_refcount.h>
+#include <AzCore/std/string/string.h>
+
+namespace AZ
+{
+    template<typename T>
+    class Interface;
+
+    class Name;
+}
+
+namespace AzFramework
+{
+    class RemoteToolsMessage : public AZStd::intrusive_refcount<size_t>
+    {
+    public:
+        AZ_CLASS_ALLOCATOR(RemoteToolsMessage, AZ::OSAllocator, 0);
+        AZ_RTTI(RemoteToolsMessage, "{8512328C-949D-4F0C-B48D-77C26C207443}");
+
+        RemoteToolsMessage()
+            : m_msgId(0)
+            , m_senderTargetId(0)
+            , m_customBlob(nullptr)
+            , m_customBlobSize(0)
+            , m_isBlobOwner(false)
+            , m_immediateSelfDispatch(false)
+        {
+        }
+        RemoteToolsMessage(AZ::u64 msgId)
+            : m_msgId(msgId)
+            , m_senderTargetId(0)
+            , m_customBlob(nullptr)
+            , m_customBlobSize(0)
+            , m_isBlobOwner(false)
+            , m_immediateSelfDispatch(false)
+        {
+        }
+
+        virtual ~RemoteToolsMessage();
+
+        //! Add a custom data blob to a Remote Tools message
+        //! @param blob Pointer to the start of the data blob
+        //! @param blobSize Size of the data blob
+        //! @param ownBlob Whether this message owns the life cycle of the blob memory
+        void AddCustomBlob(const void* blob, size_t blobSize, bool ownBlob = false);
+
+        //! Sets if this message can be immediately dispatched if the receipient is itself
+        //! @param immediateSelfDispatchEnabled Whether this message can be immediately dispatched
```

Comment:
Seeing this method made me wonder if there's a reason to hold off on immediate dispatch if it's just looping back to yourself? If it's appropriate, it would be nice for the comment to include why one would use true or false.

Reply:
Hmm, I think this can actually go away. I don't believe ImmediateSelfDispatchEnabled has a usage anymore.